rotkehlchen/chain/bitcoin/transactions.py

```diff
@@ -0,0 +1,226 @@
+                # example tx 519c79dc444009744e5dec35817d0a7530ba3601aed58ac34837397134177db8
+                # TODO: figure out how to handle this and other script types that are not
+                #  properly decoded by the current logic
+                print('------- no address --------')
+                print(tx_in_out)
+
+        event_identifier = f'btc_{tx.tx_id}'
+        if event_type == HistoryEventType.SPEND:  # Create fee events
+            total_input = sum(per_address_totals.values())
+            for address, amount in per_address_totals.items():
+                events.append(HistoryEvent(
+                    event_identifier=event_identifier,
+                    sequence_index=(sequence_index := sequence_index + 1),
+                    timestamp=tx.timestamp,
+                    location=Location.BITCOIN,
+                    event_type=HistoryEventType.SPEND,
+                    event_subtype=HistoryEventSubType.FEE,
+                    asset=A_BTC,
+                    amount=(share := (amount / total_input) * tx.fee),
+                    notes=f'Burn {share} BTC for gas',
+                    location_label=address,
+                ))
+                # Subtract fee share from the address's total input amount
+                per_address_totals[address] -= share
+
+        for address, total_amount in per_address_totals.items():
+            events.append(HistoryEvent(
+                event_identifier=event_identifier,
+                sequence_index=(sequence_index := sequence_index + 1),
+                timestamp=tx.timestamp,
+                location=Location.BITCOIN,
+                event_type=event_type,
+                event_subtype=HistoryEventSubType.NONE,
+                asset=A_BTC,
+                amount=FVal(total_amount),
+                location_label=address,
+            ))
+
+    return events
+
+
+def query_bitcoin_account_transactions(
+        database: 'DBHandler',
+        from_timestamp: Timestamp,
+        to_timestamp: Timestamp,
+        account: BTCAddress,
+) -> None:
+    """Query bitcoin account transactions for the specified time range,
+    decode them into HistoryEvents, and save the results to the db.
+    """
+    log.debug(f'Querying transactions for bitcoin account {account}')
+    events_db = DBHistoryEvents(database)
+    with database.conn.read_ctx() as cursor:
+        last_tx_id = database.get_dynamic_cache(
+            cursor=cursor,
+            name=DBCacheDynamic.LAST_BITCOIN_TX_ID,
+            address=account,
+        )
+
+    if len(tx_list := _query_tx_list_from_api(
+        account=account,
+        from_timestamp=from_timestamp,
+        to_timestamp=to_timestamp,
+        existing_tx_id=last_tx_id,
+    )) == 0:
+        log.debug(f'No new transactions found for bitcoin account {account}')
+        return
+
+    events = []
+    for tx in tx_list:
+        events.extend(decode_history_events_for_bitcoin_tx(tx))
```

Comment:
hmmm. but this will contain events that does not affect the tracked address(`account`).